src/meltano/core/discovery_file.py

```diff
@@ -0,0 +1,44 @@
+"""Legacy discovery file object."""
```

Comment:
I had to move the `DiscoveryFile` object to a separate module to avoid circular imports.